File: api-config.js
(function () {
  try {
    // نفس منفذ الصفحة والخادم — لا حاجة لإدخال عنوان يدوياً
    var o = window.location && window.location.origin;
    window.TERYAQI_API_BASE = o && o !== 'null' ? o : 'http://localhost:8080';
  } catch (e) {
    window.TERYAQI_API_BASE = 'http://localhost:8080';
  }
})();

File: flutter_app/lib/config/api_config.dart
import 'package:flutter/foundation.dart';
const String kAndroidEmulatorApiBase = 'http://10.0.2.2/Teryaqi-main';

const String _apiBaseFromEnv = String.fromEnvironment(
  'TERYAQI_API_BASE',
  defaultValue: '',
);

String defaultApiBaseUrl() {
  if (kIsWeb) return 'http://localhost:8080';
  if (defaultTargetPlatform == TargetPlatform.android) {
    final fromEnv = _apiBaseFromEnv.trim();
    if (fromEnv.isNotEmpty) {
      return fromEnv.replaceAll(RegExp(r'/$'), '');
    }
    return kAndroidEmulatorApiBase;
  }
  return 'http://127.0.0.1:8080';
}
